### PR TITLE
docs: update structure for `Wallet` docs

### PR DESCRIPTION
**Pull request by dschlabach**
**What changed? Why?**
This PR updates the structure for our documentation to make it easier for developers to scan through available props and sub-components.

Adds: 
- Anatomy section
- API reference section (with props/types table)

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: site/docs/components/PropsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                {Boolean(prop.typeSimple) && Boolean(prop.type) && (
+                  <Popover.Root>
+                    <Popover.Trigger>
+                      <InfoIcon className="size-[14px] text-[#4c4c4c] dark:text-stone-400" />
+                    </Popover.Trigger>
+                    <Popover.Content
+                      side="top"
+                      align="center"
+                      className="data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50 mx-2 rounded-md border border-gray-200 bg-white p-3 text-sm shadow-lg data-[state=closed]:animate-out data-[state=open]:animate-in dark:border-stone-800 dark:bg-stone-900"
+                    >
```
Comment (alessey) [inline site/docs/components/PropsTable.tsx:72]: nit: extract to const to reuse above

File: site/docs/components/svg/infoIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const InfoIcon = ({ className }: { className?: string }) => (
+  <svg
+    xmlns="http://www.w3.org/2000/svg"
+    viewBox="0 0 24 24"
+    fill="none"
+    stroke="currentColor"
+    strokeWidth="2"
+    strokeLinecap="round"
+    strokeLinejoin="round"
+    className={cn(className)}
```
Comment (alessey) [inline site/docs/components/svg/infoIcon.tsx:12]: may want to use role="img" and aria-labelledby + id on your title to help screen readers

File: site/docs/pages/wallet/wallet.mdx [hunk 3]
```diff
@@ -61,6 +62,147 @@ export function WalletComponents() {
...
+Serves as the main container for all wallet-related components.
+
+<PropsTable data={[{
+  name: 'children',
+  typeSimple: 'React.ReactNode',
+description: 'Content of the `Wallet` component',
+}, ]} />
+
+### ConnectWallet
+Handles the wallet connection process. Place child components inside to customize the connect button appearance.
```
Comment (alessey) [inline site/docs/pages/wallet/wallet.mdx:113]: we need to standardize on what we call the "child components".  in other docs we refer to them as subcomponents, and children below references "customized content"

Comment (dschlabach) [reply site/docs/pages/wallet/wallet.mdx:113]: I am in support of standardizing around the following:
1. Components that we ship are _subcomponents_ (e.g. ConnectWallet, NFTMedia, etc)
2. children the user passes in are just `children`

**Comment by mindapivessa** [comment]
**Structure**
- Pre-requisites
Before using the` Wallet`component, ensure you've completed the [Getting Started](https://onchainkit-git-dms-docs-structure-coinbase-vercel.vercel.app/getting-started) steps.

- Quickstart
1. Set up your wallet connections
Configure the wagmi provider and update the `appName` as that will be displayed to the user when they connect their wallet.
2. Import the component
//code snippet with imports + WalletDefault
- Anatomy
- API reference
- Customization
- Usage
With Sign In With Ethereum (SIWE)




**UI and copies**
- "Quick start" -> "Quickstart" (update these for other components too
- Use #3c393f for table divider (matching it with the rest of the page)
- Use pb-2 for the column headers (now it's pb-1)

**Comment by dschlabach** [comment]
> **UI and copies**
> 
> * "Quick start" -> "Quickstart" (update these for other components too
> * Use #3c393f for table divider (matching it with the rest of the page)
> * Use pb-2 for the column headers (now it's pb-1)

Made these changes, responded to comments in doc on other points

FYI i think it's #ececec instead of #3c393f for the borders in vocs

CleanShot 2024-11-18 at 15 16 06@2x

**Pull request closed without merging**